zebra-scan/src/service/scan_task/executor.rs

```diff
@@ -0,0 +1,47 @@
+//! The scan task executor
+
+use futures::{stream::FuturesUnordered, FutureExt, StreamExt};
+use tokio::{
+    sync::mpsc::{UnboundedReceiver, UnboundedSender},
+    task::JoinHandle,
+};
+use tracing::Instrument;
+use zebra_chain::BoxError;
+
+use super::scan::ScanRangeTaskBuilder;
+
+pub fn spawn_init() -> (
+    UnboundedSender<ScanRangeTaskBuilder>,
+    JoinHandle<Result<(), BoxError>>,
+) {
+    let (scan_task_sender, scan_task_receiver) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
Question: Why unbounded is used ? I think that if we use unbounded channels we have to limit the number of messages somewhere else.

Reply:
That's a good point. Fixed in https://github.com/ZcashFoundation/zebra/pull/8222/commits/217fcda3172c9dd7b502247a98aa5664601d2b85. The `std::mpsc::channel` used in `ScanTask` is also unbounded, but I left a TODO there to move that logic to a tower service (maybe the scan service).